cas-server-support-osf/src/test/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsActionTests.java

```diff
@@ -0,0 +1,86 @@
+package io.cos.cas.authentication.handler.support;
+
+import io.cos.cas.AbstractTestUtils;
+import io.cos.cas.adaptors.postgres.types.DelegationProtocol;
+import io.cos.cas.authentication.OpenScienceFrameworkCredential;
+import io.cos.cas.mock.MockOsfRemoteAuthenticateAction;
+import org.jasig.cas.CentralAuthenticationService;
+import org.jasig.cas.authentication.Authentication;
+import org.jasig.cas.ticket.TicketGrantingTicket;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.test.MockRequestContext;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class tests the {@link OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction} class.
+ *
+ * @author Longze Chen
+ * @since  4.1.5
+ */
+public class OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsActionTests {
+
+    private static final String TICKET_GRANTING_TICKET_ID
+            = "TGT-00-xxxxxxxxxxxxxxxxxxxxxxxxxx.cas0";
+
+    private static final String PAC4J_DELEGATION_PROFILE_ID = "MockCasProfile#0001-1234-5678";
+
+    @Test
+    public void verifyInstitutionSamlShibbolethFlow() throws Exception {
+        assertTrue(Boolean.TRUE);
+    }
+
+    @Test
+    public void verifyInstitutionCasPac4jFlow() throws Exception {
```

Comment:
For institution login by CAS pac4j, verify the final `credential` and the return event.
TODO: add check that `TGT` is destroyed.